testslide/matchers.py

```diff
@@ -265,7 +313,7 @@ def __init__(self) -> None:
 
 class RegexMatches(Matcher):
     """
-    Compares true if other mathes given regex.
+    Compares true if other matches given regex.
     """
 
     def __init__(self, pattern: str, flags: int = 0) -> None:
```

Comment:
**What I would like to mention is that, adding a type check here was the reason the build was failing. I wonder why was it so. And hence I couldn't add a type check here. Any comments on this will be appreciated.**

Reply:
I'm not sure I'm following you ... what failure specifically are you referring to?

Checking the history of your commits I see that the code was something like
```python
def __init__(self, pattern: str, flags: int = 0) -> None:
    if not isinstance(pattern, str) or not isinstance(flags, int):
        raise ValueError(
            f"RegexMatches(...) expects 'str' and 'int' as arguments while '{type(pattern).__name__}' and '{type(flags).__name__}' were provided"
        )
    ...
```
how is this causing random test failures?

---

Side (and personal) note: to make the code-review easier I would suggest to limit merging master branch in your branch.
This because it makes harder to trace changes and makes the history more confusing. Clearly, this does not mean that you should not be doing it but do it parsimoniously.
I do tend to use `git rebase` on the PR branch if the PR takes a very long time to be reviewed and merged in order to gather all the updates.
If a PR is short-living it would be unlikely to have failures caused by other changes and anyway when github actions runs tests they are run on top of your branch merged with master, so tests will detect if your changes would cause test failures caused by new code.

Reply:
I wanted to point the PR checks. 
as mentioned by `deathowl` [here](https://github.com/facebook/TestSlide/pull/292#issuecomment-806680047)

I was curious why the build was failing so i checked for the build logs as i mentioned [here](https://github.com/facebook/TestSlide/pull/292#issuecomment-806760133)

[This](https://github.com/facebook/TestSlide/pull/292/checks?check_run_id=2198950761) is the failure i was mentioning about.

And, I'll remember the note.

Reply:
In fact all of the commits were failing build test, until the last one.